framework/Optimizers/stepManipulators/GradientHistory.py

```diff
@@ -70,6 +70,17 @@ def getInputSpecification(cls):
               stochastic surfaces or low-order gradient approximations such as SPSA, a small value
               for the shrink factor is recommended. If an optimization path appears to be converging
               early, increasing the shrink factor might improve the search. \default{1.15} """))
+    specs.addSub(InputData.parameterInputFactory('terms', contentType=InputTypes.IntegerType,
+        descr=r"""the number of previous gradient evaluations to include when determining a new step
```

Comment:
I wish if we use a different name for 'terms', it might confuse users to think that number of terms has to do with the order of the derivate, meaning 2 terms as in Taylor's expansion. I know that gradient should only mean first-order derivatives but again some confuse gradient with derivatives. Sure, the manual removes the confusion, but more than anyone you know that many of the users do not pay enough attention to it. Suggestion: 'prevGrads'. You do not have to change this, just giving a suggestion.

Reply:
After polling some users, it looks like `<window>` is a winner.